planner/core/rule_join_reorder.go

```diff
@@ -89,6 +126,18 @@ func (s *joinReOrderSolver) optimizeRecursive(ctx sessionctx.Context, p LogicalP
 		if err != nil {
 			return nil, err
 		}
+		sort.Slice(leftJoinTabs, func(i, j int) bool {
+			if len(leftJoinTabs[j].eqCond) > 0 && leftJoinTabs[i].p.Schema().ColumnIndex(leftJoinTabs[j].eqCond[0].GetArgs()[0].(*expression.Column)) >= 0 {
+				return true
+			}
+			if len(leftJoinTabs[i].eqCond) > 0 && len(leftJoinTabs[j].eqCond) == 0 {
+				return true
+			}
+			return false
```

Comment:
Shall we compare the row count of these `leftJoinTabs` as well when deciding the order, i.e, if `leftJoinTabs[i].p.stats.RowCount` is smaller than `leftJoinTabs[j].p.stats.RowCount`, we put j after i?